lib/queuePopulator/KafkaLogConsumer/ListRecordStream.js

```diff
@@ -50,6 +50,29 @@ class ListRecordStream extends stream.Transform {
         }
     }
 
+    /**
+     * Get object metadata from oplog event
+     * @param {ChangeStreamDocument} changeStreamDocument changeStreamDocument
+     * @returns {string|undefined} stringified objectMd
+     */
+    _getObjectMd(changeStreamDocument) {
+        // Metadata values from updateDescription are used instead of
+        // fullDocument as they represent the exact changes that occured
+        // to the object. We only need this for the update operation.
+
+        // The fullDocument value contains a copy of the entire updated
+        // document at a point in time after the update, this can cause issues
+        // when the object gets deleted just after the update (only for updates).
+        const updateDescriptionMd = changeStreamDocument.updateDescription
+            && changeStreamDocument.updateDescription.updatedFields &&
+            changeStreamDocument.updateDescription.updatedFields.value;
+        if (changeStreamDocument.operationType === 'update') {
+            return updateDescriptionMd && JSON.stringify(updateDescriptionMd);
```

Comment:
are there not 2 cases, e.g. when we set the whole value vs when we update a single 'deep' field ?

From yesterday, i understand we update single field only when setting the 'delete' flag, so we may actually just skip that oplog entry?

Reply:
it will be skipped in the logReader level

Reply:
https://github.com/scality/backbeat/blob/d9aa80bfcc4335f65379223e4226c504da16ce52/lib/queuePopulator/LogReader.js#L496-L498

Reply:
may be useful to add a comment in `getObjectMD()` explaining what we expect here (and how it will be fixed otherwise)